src/core/include/openvino/op/util/evaluate_helpers.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/partial_shape.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+
+/**
+ * @brief Get the tensors shapes as ov::PartialSHape
+ *
+ * @param tensors  Input tensors vector to get its shapes.
+ * @return Vector of partial shapes sam size as input tensor vector.
+ */
+std::vector<PartialShape> get_tensors_shapes(const TensorVector& tensors);
```

Comment:
Why do we need this function? Is it a common case for operations?
Moreover `ov::Tensor` doesn't contain `ov::PartialShape` and has only `ov::Shape`.

Reply:
It is common when `shape_infer` used for output shape calculations. It helps get shapes as partial shape instead manual rewrite tensor shapes into vector and also in case when op has variadic inputs count it collect all of them in same way.

Will rename it to  `get_tensors_partial_shapes` to explicitly give information what is the return value. When accepted in one PR and merged then propagate this change for others.

Function `shape_infer` not working on `ov::Shape` as this shape has not compatible interface with `ov::PartialShape`.